tensorflow_datasets/testing/dataset_builder_testing.py

```diff
@@ -261,6 +272,20 @@ def test_download_and_prepare_as_dataset(self):
     else:
       self._download_and_prepare_as_dataset(self.builder)
 
+    if not self.SKIP_CHECKSUMS:
+      with self._subTest("url_checksums"):
+        self._test_checksums()
+
+  def _test_checksums(self):
+    filepath = os.path.join(checksums._get_path(self.builder.name))
```

Comment:
Note you should only check the file existence if the `dl_manager.download` has been called. If there is no check to do if `self._download_urls` is empty.